src/main/java/uk/ac/cam/cl/dtg/segue/api/GlossaryFacade.java

```diff
@@ -0,0 +1,167 @@
+import javax.ws.rs.core.Request;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.xml.bind.annotation.XmlElement;
+import java.io.IOException;
+import java.util.*;
+
+import static com.google.common.collect.Maps.immutableEntry;
+import static uk.ac.cam.cl.dtg.segue.api.Constants.*;
+
+/**
+ * Glossary Facade
+ *
+ * This facade is intended to provide access to glossary terms.
+ *
+ */
+@Path("/glossary")
+@Api(value = "/glossary")
+public class GlossaryFacade extends AbstractSegueFacade {
+    private static final Logger log = LoggerFactory.getLogger(GlossaryFacade.class);
+
+    private final IContentManager contentManager;
+    private final String contentIndex;
+
+    /**
+     * @param properties     - to allow access to system properties.
+     * @param contentManager - so that metadata about content can be accessed.
+     * @param contentIndex   - to access the right version of the content.
+     * @param logManager     - for logging events using the logging api.
+     */
+    @Inject
+    public GlossaryFacade(final PropertiesLoader properties, final IContentManager contentManager,
+                          @Named(CONTENT_INDEX) final String contentIndex,
+                          final ILogManager logManager) {
+        super(properties, logManager);
+        this.contentManager = contentManager;
+        this.contentIndex = contentIndex;
+    }
+
+    /**
+     * Gets all the glossary terms that are indexed.
+     *
+     * @param limit      - Maximum amount of terms to retrieve. Used for pagination.
+     * @param startIndex - Index from which to start retrieving when results exceed limit.
+     *
+     * @return Paginated list of glossary terms.
+     */
+    @GET
+    @Path("terms")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Get all the glossary terms that are indexed.")
+    public final Response getTerms(@QueryParam("start_index") final String startIndex,
+                                   @QueryParam("limit") final String limit) {
+
+        Map<Map.Entry<BooleanOperator, String>, List<String>> fieldsToMatch = Maps.newHashMap();
+        fieldsToMatch.put(immutableEntry(BooleanOperator.AND, TYPE_FIELDNAME), Collections.singletonList("glossaryTerm"));
+
+        ResultsWrapper<ContentDTO> c;
+        try {
+            Integer resultsLimit = null;
+            Integer startIndexOfResults = null;
+
+            if (null != limit) {
+                resultsLimit = Integer.parseInt(limit);
+            } else {
+                resultsLimit = DEFAULT_RESULTS_LIMIT;
+            }
+
+            if (null != startIndex) {
+                startIndexOfResults = Integer.parseInt(startIndex);
+            } else {
+                startIndexOfResults = 0;
+            }
+
+            c = this.contentManager.findByFieldNames(this.contentIndex, fieldsToMatch, startIndexOfResults, resultsLimit);
+            return Response.ok(c).build();
```

Comment:
Probably worth adding an etag cache for this endpoint if it is going to be hit many times as it will not change between content version changes.